contracts/interchain-token-service/src/contract.rs

```diff
@@ -186,52 +113,61 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
             .into()
     }
 
-    fn deploy_remote_interchain_token(
-        env: &Env,
-        caller: Address,
-        destination_chain: String,
-        _token_id: String,
-        gas_token: Token,
-    ) {
-        let destination_address = String::from_str(env, "");
-
-        // TODO: abi encode with MessageType.DeployInterchainToken
-        let payload = bytes!(env,);
+    fn deploy_interchain_token(
+        _env: &Env,
+        _caller: Address,
+        _salt: BytesN<32>,
+        _destination_chain: String,
+        _token_metadata: TokenMetadata,
+        _minter: Option<Bytes>,
+        _gas_token: Token,
+    ) -> Result<BytesN<32>, ContractError> {
+        todo!()
+    }
 
-        Self::pay_gas_and_call_contract(
-            env.clone(),
-            caller,
-            destination_chain,
-            destination_address,
-            payload,
-            gas_token,
-        );
+    fn deploy_remote_interchain_token(
+        _env: &Env,
+        _caller: Address,
+        _salt: BytesN<32>,
+        _minter: Option<Bytes>,
+        _destination_chain: String,
+        _gas_token: Token,
+    ) -> Result<BytesN<32>, ContractError> {
+        // TODO: implementation
+        // deploy salt
+        // minter approval
+        // registered token address - get token metadata
+        // abi encode with MessageType.DeployInterchainToken
+        // token id
+        // pay_gas_and_call_contract
```

Comment:
these comments should be in the task or some other place instead of the code itself